Clock.py

```diff
@@ -30,61 +30,134 @@
 Cuckoo clock substitute.
 
 Serial port, code speed, and audio preferences should be specified by running the
-'configure.sh' script or executing 'python3 Configure.py'.
+"configure.sh" script or executing "python3 Configure.py".
 """
 
+NUMBER = [ "oh", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten", "eleven", "twelve"]
+
+#
+# Convert hours/minutes/seconds since midnight to seconds since midnight:
+#
+def hms_to_seconds(hours, minutes, seconds):
+    return hours * 3600 + minutes * 60 + seconds
+
+#
+# Convert from numeric representation to text ("9" to "nine")
+#
+def number_to_text(n):
+    if (n < 0) or ( n > 12):
+        return ""
+    return NUMBER[n]
+#
+# Convert 0-24 hours to 0-12 hours by subtracting 12 if needed:
+#
+def truncate_hours(h):
+    if h <= 12:
+        return h        # 0-12 is returned as-is
+    else:
+        return h % 12   # 13-on is truncated to 1-on
+
+#
+# Convert seconds since midnight to corresponding clock hour
+#
+def clock_hour(t):
+    return int(t / 3600)
+
+#
+# Convert seconds since midnight to the corresponding clock minute
+#
+def clock_minutes(t):
+    return int(int(t % 3600) / 60)
+
+#
+# Take an starting point and round up to the next interval-multiple:
```

Comment:
# Take a starting point and round up to the next interval-multiple:

Reply:
Got it - fixed.